src/internal.rs

```diff
@@ -573,6 +582,11 @@ pub(crate) struct Rav1dFrameData {
 }
 
 impl Rav1dFrameData {
+    pub fn bd_fn(&self) -> &'static Rav1dFrameContext_bd_fn {
+        let bpc = BPC::from_bitdepth_max(self.bitdepth_max);
```

Comment:
Constructing a bpc seems like it will do unnecessary work here. Can we just have a check against a constant here?

Reply:
That's what's happening.  `BPC::from_bitdepth_max` checks against a `const` (`BitDepth8::new(())`) to construct `BPC`, and `BPC` is basically an index.  It compiles down to a `cmp` and `jne` to the two statically known `fn`s.